src/Storages/StorageBuffer.h

```diff
@@ -160,7 +160,14 @@ friend class BufferBlockOutputStream;
     void writeBlockToDestination(const Block & block, StoragePtr table);
 
     void backgroundFlush();
-    void reschedule();
+
+    /// try_to_lock here for direct flush,
```

Comment:
Maybe `skip_locked`? By the way, can we always skip the locked buffers? They are either being flushed so we don't need to schedule them again, or being inserted and will be rescheduled after the insert completes.

Reply:
>Maybe skip_locked?

Indeed, sounds better (my naming is based on the std::unique_lock header)

>By the way, can we always skip the locked buffers?

This looks completely find indeed, at first I wrote like this, but later decide to not use try lock in case of background flush,  but thinking about this more, and it either should not call reschedule after flushing each layer or also use try lock

Reply:
>and it either should not call reschedule after flushing each layer or also use try lock

Actually background call reschedule only once, everything is OK, so let's use try_to_lock for both code path, agree